lib/identity_cache/cached/primary_index.rb

```diff
@@ -54,7 +54,10 @@ def cache_key(id)
 
       def load_one_from_db(id)
         record = build_query(id).take
-        model.send(:setup_embedded_associations_on_miss, [record]) if record
+        if record
+          model.send(:setup_embedded_associations_on_miss, [record])
+          record.send(:mark_as_loaded_by_idc)
+        end
```

Comment:
Should we do the same thing for `load_multi_from_db`?

Reply:
Here you call the `mark_as_loaded_by_idc` in the `load_one_from_db` method is based on the assumption that whenever the method is called, the IDC is filled, right?